### PR TITLE
Ports TGs admin only supermatter spiders

### PR DESCRIPTION
**Pull request by Xoxeyos**
# Document the changes in your pull request

Created by [capsaicinz](https://github.com/capsaicinz); Adds little creatures that will aggressively hunt down a target and dust it, killing itself in the process. tend to enjoy hunting in swarms, for sport mostly.

# Changelog

Adds admin-spawn only Super Matter spiders. Horrifying, yet hilarious. Created by capsaicinz.

:cl: capsaicinz, Xoxeyos, arcturus-prime
rscadd: Adds supermatter spiders.
/:cl:

### CODE REVIEW

**Comment by KillerOrcaCora** [comment]
Sponsored and brought to you by capsaicinz

**Review by adamsong — DISMISSED** [review]
Add a way to spawn them

**Comment by KittyNoodle** [comment]
Adam you know you can just use the game panel’s spawn mob right

**Review by nesisknesis — APPROVED** [review]

**Comment by Xoxeyos** [comment]
> Add a way to spawn them

Going to start working on it now.

**Comment by JamieD1** [comment]
Dusting is one of the only ways you cant be revived from.

Im not sure if admins can control themselves, can test it.

**Comment by Xoxeyos** [comment]
> Add a way to spawn them

Done, I guess I forgot to build it.

**Comment by JamieD1** [comment]
Add this to a smite, this should be used VERY rarely

**Comment by JamieD1** [comment]
> Add this to a smite, this should be used VERY rarely

You thumbs up'd it but didnt do what I suggested :D

**Comment by ToasterBiome** [comment]
@Xoxeyos

**Comment by Xoxeyos** [comment]
> @Xoxeyos

Finishing it up soon, I've been busy this past week.

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
 				to_chat(target, span_reallybigphobia("HENK!! HENK!! HENK!! YOU DID SOMETHING EXTREMELY DUMB, AND MADE GOD MAD. CRY ABOUT IT."))
 			var/mob/living/carbon/human/H = target
 			H?.cluwneify()
+		if(ADMIN_PUNISHMENT_SMSPIDER)
+			var/confirm = alert(usr, "Dust target with a spider?", "Confirm? There is no chance of revival!", "Yes", "No")
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion
			var/confirm = alert(usr, "Dust target with a spider? There is no chance of revival!", "Supermatter Spider", "Yes", "No")
```

2nd argument is the question, third is the title of the window that pops up.

File: code/modules/admin/verbs/randomverbs.dm [hunk 2]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+			if(confirm == "No")
+				return
+			//What's an open turf within the target's sight?
+			var/list/FOVlist = oview(target,5)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion
			var/FOVlist = circleviewturfs(target,5)
```

File: code/modules/admin/verbs/randomverbs.dm [hunk 3]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+			//Let's compile a list of these turfs.
+			var/list/FOVTurflist = new/list()
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion

```

File: code/modules/admin/verbs/randomverbs.dm [hunk 4]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+			//This adds all available open turfs into this list.
+			for(var/turf/open/item in FOVlist)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion

```

File: code/modules/admin/verbs/randomverbs.dm [hunk 5]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+				FOVTurflist.Add(item)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion

```

File: code/modules/admin/verbs/randomverbs.dm [hunk 6]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+
+			//Okay, now we spawn a spider on the turf picked, and have it target the victim.
+			var/turf/open/OV = pick(FOVTurflist)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion

```

File: code/modules/admin/verbs/randomverbs.dm [hunk 7]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+			var/mob/living/loser/ = new /mob/living/simple_animal/hostile/smspider(OV)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion
			var/mob/living/simple_animal/hostile/smspider/spider = new /mob/living/simple_animal/hostile/smspider(pick(FOVlist))
```

File: code/modules/admin/verbs/randomverbs.dm [hunk 8]
```diff
@@ -1190,6 +1191,23 @@ Traitors and the like can also be revived with the previous role mostly intact.
...
+			loser.attack_animal(target)
```
Comment (ToasterBiome) [inline code/modules/admin/verbs/randomverbs.dm]:
```suggestion
			spider.GiveTarget(target)
```

**Review by ToasterBiome — APPROVED** [review]

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
fuck i pressed the wrong button

**Comment by Xoxeyos** [comment]
> Add a way to spawn them

done, the prioritize the target, ignore non-targets, and if stuck in  5x5 walls with a hole in the center they prioritize the center, then cardinals, and then diagonals

**Review by ToasterBiome — APPROVED** [review]

**Comment by Xoxeyos** [comment]
@adamsong

**Review by JamieD1 — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**